From: Ben Hutchings <ben@decadent.org.uk>
Subject: iwlwifi: Do not request unreleased firmware
Bug-Debian: http://bugs.debian.org/689416
Bug-Debian: http://bugs.debian.org/705655

The iwlwifi driver currently supports firmware API versions 4-6 for
IWL6000/6300 devices.  It will request the file for the latest
supported version and then fall back to earlier versions.  However,
the latest version that has actually been released is 4, so we expect
the requests for versions 6 and then 5 to fail.

Similarly for IWL6005/6205 devices it requests API version 6, which is
available but wasn't released until long after this version of the
driver (and after the wheezy freeze).

The installer appears to report any failed request, and it is probably
not easy to detect that this particular failure is harmless.  So stop
requesting the unreleased firmware.

--- a/drivers/net/wireless/iwlwifi/iwl-6000.c
+++ b/drivers/net/wireless/iwlwifi/iwl-6000.c
@@ -46,9 +46,10 @@
 #include "iwl-cfg.h"
 
 /* Highest firmware API version supported */
-#define IWL6000_UCODE_API_MAX 6
+#define IWL6000_UCODE_API_MAX 4 /* v5-6 are supported but not released */
+#define IWL6000G2_UCODE_API_MAX 5 /* v6 was released too late */
 #define IWL6050_UCODE_API_MAX 5
-#define IWL6000G2_UCODE_API_MAX 6
+#define IWL6000G2B_UCODE_API_MAX 6
 #define IWL6035_UCODE_API_MAX 6
 
 /* Oldest version we won't warn about */
@@ -413,7 +414,7 @@ struct iwl_cfg iwl6005_2agn_d_cfg = {
 
 #define IWL_DEVICE_6030						\
 	.fw_name_pre = IWL6030_FW_PRE,				\
-	.ucode_api_max = IWL6000G2_UCODE_API_MAX,		\
+	.ucode_api_max = IWL6000G2B_UCODE_API_MAX,		\
 	.ucode_api_ok = IWL6000G2B_UCODE_API_OK,		\
 	.ucode_api_min = IWL6000G2_UCODE_API_MIN,		\
 	.eeprom_ver = EEPROM_6030_EEPROM_VERSION,		\
